api/src/ddoc-extraction.js

```diff
@@ -19,20 +19,52 @@ var getCompiledDdocs = function(callback) {
   });
 };
 
-var isUpdated = function(settings, ddoc, callback) {
-  db.medic.get(ddoc._id, function(err, oldDdoc) {
+var areAttachmentsEqual = (oldDdoc, newDdoc) => {
+  if (!oldDdoc._attachments && !newDdoc._attachments) {
+    // no attachments found
+    return true;
+  }
+  if (!oldDdoc._attachments || !newDdoc._attachments) {
+    // one ddoc has attachments and the other doesn't
+    return false;
+  }
+  if (Object.keys(oldDdoc._attachments).length !== Object.keys(newDdoc._attachments).length) {
+    // one ddoc has more attachments than the other
+    return false;
+  }
+  // check all attachment data
+  return Object.keys(oldDdoc._attachments).every(name => {
+    return newDdoc._attachments[name] &&
+           newDdoc._attachments[name].data === oldDdoc._attachments[name].data;
+  });
+};
+
+var isUpdated = function(settings, newDdoc, callback) {
+  db.medic.get(newDdoc._id, { attachments: true }, function(err, oldDdoc) {
     if (err && err.error !== 'not_found') {
       return callback(err);
     }
-    ddoc._rev = oldDdoc && oldDdoc._rev;
-    if (ddoc._id === CLIENT_DDOC_ID) {
-      ddoc.app_settings = settings;
+    // set the rev so we can update if necessary
+    newDdoc._rev = oldDdoc && oldDdoc._rev;
+    if (newDdoc._id === CLIENT_DDOC_ID) {
+      newDdoc.app_settings = settings;
+    }
+    if (!oldDdoc) {
+      // this is a new ddoc - definitely install it
+      return callback(null, newDdoc);
+    }
+    if (!areAttachmentsEqual(oldDdoc, newDdoc)) {
+      // attachments have been updated - install it
+      return callback(null, newDdoc);
     }
-    if (oldDdoc && _.isEqual(ddoc, oldDdoc)) {
-      // unmodified
+
+    // we've changed attachment data so we know they're identical where it counts
+    oldDdoc._attachments = newDdoc._attachments;
```

Comment:
Afraid that, when neither ddocs have `_attachments`, this line assigns `oldDdoc._attachments` with `undefined` as value. As a consequence, the following `_.isEqual` call always fails and the ddoc is always updated.

This is what test_api_integration looks like now:
```
remove-empty-parents migration
Updating docs: _design/medic-client,_design/medic-conflicts,_design/medic-scripts,_design/medic-sms
    ✓ should not affect well-defined parents (1643ms)
Updating docs: _design/medic-client,_design/medic-conflicts,_design/medic-scripts,_design/medic-sms
    ✓ should delete parents which are null (1624ms)
Updating docs: _design/medic-client,_design/medic-conflicts,_design/medic-scripts,_design/medic-sms
    ✓ should delete parents which have no properties (1600ms)
```

Reply:
Good spotting! Fixed. Will check the build log before merging.